src/backend/distributed/utils/node_metadata.c

```diff
@@ -87,21 +93,45 @@ master_add_node(PG_FUNCTION_ARGS)
 	bool hasMetadata = false;
 	bool nodeAlreadyExists = false;
 
-	Datum returnData = AddNodeMetadata(nodeNameString, nodePort, groupId, nodeRack,
-									   hasMetadata, &nodeAlreadyExists);
+	Datum nodeRecord = AddNodeMetadata(nodeNameString, nodePort, groupId, nodeRack,
+									   hasMetadata, NODE_STATE_INACTIVE,
+									   &nodeAlreadyExists);
 
 	/*
-	 * After adding new node, if the node is not already exist, we  replicate all existing
-	 * reference tables to the new node. ReplicateAllReferenceTablesToAllNodes replicates
-	 * reference tables to all nodes however, it skips nodes which already has healthy
-	 * placement of particular reference table.
+	 * After adding new node, if the node is not already exist, we will activate
+	 * the node. This means we will replicate all reference tables to the new
+	 * node.
 	 */
 	if (!nodeAlreadyExists)
 	{
-		ReplicateAllReferenceTablesToAllNodes();
+		nodeRecord = ActivateNode(nodeNameString, nodePort);
 	}
 
-	PG_RETURN_CSTRING(returnData);
+	PG_RETURN_CSTRING(nodeRecord);
+}
+
+
+/*
+ * master_prepare_node function adds a new node to the cluster as inactive node
```

Comment:
Chatted with Ozgun last week and he was OK with changing back to `master_add_inactive_node`  if we're OK with it ("if you and Burak are good with it, I'm good with it"). I said it was more intuitive to me that `master_add_inactive_node` + `master_activate_node` = `master_add_node`.

I'd like to change `master_prepare_node` to `mamaster_add_inactive_node`; can we go ahead with that?